storage/build.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use semver::Version;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+fn main() {
+    let crate_version = Version::parse(VERSION).expect("failed to parse CARGO_PKG_VERSION");
```

Comment:
We need to say what the bad string was, where it came from and why it's a problem.

```suggestion
    let crate_version = Version::parse(VERSION).expect(format!("The version '{}' in Cargo.toml does not match the semver spec", VERSION));
```

Reply:
As mentioned offline, I will add a comment here. 

In a well-formed environment, it should be impossible for cargo to provide a value that does not parse as the value provided was previously a valid `semver::Version`:

https://github.com/rust-lang/cargo/blob/1d71929823e99bb7175098adefdedd928006c321/src/cargo/core/compiler/compilation.rs#L367

https://github.com/rust-lang/cargo/blob/1d71929823e99bb7175098adefdedd928006c321/src/cargo/core/package.rs#L151-L154

After seeing that code, I could also switch to using `CARGO_PKG_VERSION_MAJOR` directly.